bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BufferedChannel.java

```diff
@@ -243,6 +243,11 @@ public long forceWrite(boolean forceMetadata) throws IOException {
 
     @Override
     public synchronized int read(ByteBuf dest, long pos, int length) throws IOException {
+        // protect negative position read
+        if (pos < 0) {
+            throw new IllegalArgumentException("Negative position pos:" + pos);
```

Comment:
`IllegalArgumentException` is an unchecked exception, can we use `IOException` and catch it in the program?

Reply:
Thanks for review. my concern on using IOException is that current code base seems not process IOException directly. I didn't check what will happen if the IOException throws outside. For this case I think the `IllegalArgumentException` is the same as origin behaviour as this is the same exception throw by jdk.